src/dsp_tools/commands/excel2json/lists_compliance_checks.py

```diff
@@ -100,7 +162,138 @@ def make_col_names(lang: str) -> set[str]:
+        logger.error(msg)
+        raise InputError(msg)
+
+
+def _check_sheet_if_any_nodes_miss_translations(
+    df: pd.DataFrame, sheet_name: str
+) -> MissingTranslationsSheetProblem | None:
+    col_endings = [str(num) for num in _get_hierarchy_nums(df.columns)]
+    col_endings.append("list")
+    languages = _get_all_languages_for_columns(df.columns)
+    all_cols = _make_columns(col_endings, languages)
+    problems = []
+    for column_group in all_cols:
+        problems.extend(_check_one_hierarchy(column_group, df))
+    if problems:
+        return MissingTranslationsSheetProblem(sheet_name, problems)
+    return None
+
+
+def _check_one_hierarchy(columns: list[str], df: pd.DataFrame) -> list[MissingNodeTranslationProblem]:
+    problems = []
+    for i, row in df.iterrows():
+        if problem := _check_one_node_for_translations(row, columns, int(str(i))):
+            problems.append(problem)
+    return problems
+
+
+def _make_columns(nums: list[str], languages: set[str]) -> list[list[str]]:
+    return [[f"{lang}_{num}" for lang in languages] for num in nums]
+
+
+def _check_one_node_for_translations(
+    row: pd.Series[Any], columns: list[str], row_index: int
+) -> MissingNodeTranslationProblem | None:
+    missing = row[columns].isna()
+    if missing.any() and not missing.all():
+        missing_cols = [str(index) for index, is_missing in missing.items() if is_missing]
+        return MissingNodeTranslationProblem(missing_cols, row_index)
+    return None
+
+
+def _check_all_excel_for_erroneous_entries(excel_dfs: dict[str, dict[str, pd.DataFrame]]) -> None:
+    problems = []
+    for filename, excel_sheets in excel_dfs.items():
+        missing_rows: list[Problem] = [
+            p
+            for sheet_name, df in excel_sheets.items()
+            if (p := _check_sheet_for_erroneous_entries(df, sheet_name)) is not None
+        ]
+        if missing_rows:
+            problems.append(ListExcelProblem(filename, missing_rows))
+    if problems:
+        msg = ListCreationProblem(problems).execute_error_protocol()
+        logger.error(msg)
+        raise InputError(msg)
+
+
+def _check_sheet_for_erroneous_entries(df: pd.DataFrame, sheet_name: str) -> ListSheetContentProblem | None:
+    preferred_lang = _get_preferred_language(df.columns)
+    preferred_cols = _get_columns_of_preferred_lang(df.columns, preferred_lang, r"\d+")
+    preferred_cols = sorted(preferred_cols)
+    preferred_cols.insert(
+        0,
+        f"{preferred_lang}_list",
+    )
+    problems = _check_for_erroneous_entries(df, preferred_cols)
+    if problems:
+        list_problems = cast(list[Problem], problems)
+        return ListSheetContentProblem(sheet_name, list_problems)
+    return None
+
+
+def _check_for_erroneous_entries(df: pd.DataFrame, columns: list[str]) -> list[NodesPerRowProblem]:
+    problems = []
+    for i, col in enumerate(columns):
+        to_check_cols = columns[i:]
```

Comment:
doesn't this do the same work multiple times?

Reply:
Not as far as I can see.